internal/annotations/webhook_verification/webhook_verification.go

```diff
@@ -0,0 +1,45 @@
+package webhook_verification
+
+import (
+	ingressv1alpha1 "github.com/ngrok/kubernetes-ingress-controller/api/v1alpha1"
+	"github.com/ngrok/kubernetes-ingress-controller/internal/annotations/parser"
+	networking "k8s.io/api/networking/v1"
+)
+
+type EndpointWebhookValidation = ingressv1alpha1.EndpointWebhookValidation
+type SecretKeyRef = ingressv1alpha1.SecretKeyRef
+
+type webhookVerification struct{}
+
+func NewParser() parser.IngressAnnotation {
+	return webhookVerification{}
+}
+
+func (wv webhookVerification) Parse(ing *networking.Ingress) (interface{}, error) {
+	provider, err := parser.GetStringAnnotation("webhook-validation-provider", ing)
+	if err != nil {
+		return nil, err
+	}
+
+	switch provider {
+	case "sns":
+		return &EndpointWebhookValidation{Provider: provider}, nil
+	}
+
+	secretName, err := parser.GetStringAnnotation("webhook-validation-secret-name", ing)
```

Comment:
should we check for if the error is `ErrMissingAnnotations` and if so throw a validation error? Or have the parser do some validation in general? This is the first of the modules I think where if you supply 1 annotation, others are required. I think in this case if you add the provider but not the secret, then we'll throw a missing annotation here which then https://github.com/ngrok/kubernetes-ingress-controller/blob/1061b7961e0d3487a0e381d7a478bacc76943813/internal/annotations/annotations.go#L64 here we'd skip over. I'm wondering if we should have a more specific "you provided X annotation but Y, and Z are also required" type of message for the user

Reply:
I'm thinking we'd probably want to handle this with a [Validating Admission Webhook](https://kubernetes.io/docs/reference/access-authn-authz/admission-controllers/#validatingadmissionwebhook) in order to actually give a better error message back to the user in real time as opposed to making them chase it down in logs? Does that seem like the right place for it to you?

Reply:
hmm, that would probably be a much better user experience. If we had that then we'd be able to give them that validation at apply time instead of at reconcile time right? Digging through logs wouldn't be fun, but we could show pretty clear/obvious k8s event warn/errors in the interim. I've heard admission webhooks are powerful but can be dangerous foot guns and feels like something we might want to look into more in the future?